### PR TITLE
Remaining test 

### PR DESCRIPTION
**Pull request by epananth**
UpdateGithubIssue test 
1)Includes Creating and Updating Issue Comment 
2)Creating new issue comment 

Some corner cases for CreateGithubIssue
1) If the subscription is not present in the database 
2) If the error is older than current time

### CODE REVIEW

File: src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs [hunk 1]
```diff
@@ -119,5 +120,171 @@ public Issue GetIssue()
...
+                    Timestamp = new DateTime(2200, 2, 1),
+                    Arguments = "ProcessPendingUpdatesAsync error",
+                    Success = false,
+                    ErrorMessage = ErrorMessage,
+                    Action = "Create a new issue comment",
+                };
+
+            Context.RepoBranchUpdateInMemory = new List<RepositoryBranchUpdateHistoryEntry>
+                {firstError , secondError};
+            Mock<Maestro.Data.Models.Subscription> subscription = new Mock<Maestro.Data.Models.Subscription>();
```
Comment (alexperovich) [inline src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs:154]: This can just be
```
new Subscription {
  Id = Guid.Parse(SubscriptionId),
  SourceRepository = "Source Repo",
  TargetRepository = "Target Repo",
}
```

There is no reason to make a mock. This is just a data object.

File: src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs [hunk 2]
```diff
@@ -119,5 +120,171 @@ public Issue GetIssue()
...
+                    Method = AnotherMethod,
+                    Timestamp = new DateTime(2200, 3, 1),
+                    Arguments = "ProcessPendingUpdatesAsync arguments",
+                    Success = false,
+                    ErrorMessage = ErrorMessage,
+                    Action = "Update the comment",
+                };
+            Context.RepoBranchUpdateInMemory = new List<RepositoryBranchUpdateHistoryEntry>
+                {firstError , secondError , thirdError};
+            Mock<Maestro.Data.Models.Subscription> subscription = new Mock<Maestro.Data.Models.Subscription>();
```
Comment (alexperovich) [inline src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs:241]: Same thing here.

File: src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs [hunk 3]
```diff
@@ -119,5 +120,171 @@ public Issue GetIssue()
...
+            GithubClient.Setup(x => x.Issue.Comment.Update(RepoId, CommentId, Capture.In(newCommentInfo)))
+                .ReturnsAsync(comment);
+            DependencyUpdateErrorProcessor errorProcessor =
+                ActivatorUtilities.CreateInstance<DependencyUpdateErrorProcessor>(Scope.ServiceProvider,
+                    Context);
+            await errorProcessor.ProcessDependencyUpdateErrorsAsync();
+            Assert.Single(newIssues);
+            Assert.Equal("DependencyUpdateError", newIssues[0].Labels[0]);
+            Assert.Contains(RepoUrl, newIssues[0].Body);
+            Assert.Contains(SubscriptionId, newIssues[0].Body);
```
Comment (alexperovich) [inline src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs:280]: This line is duplicated.

File: src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs (hunk 1 above)
Comment (epananth) [reply src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs:154]: I changed that in createGithubIssue file and missed it here. Changing it.

File: src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs [hunk 4]
```diff
@@ -119,5 +120,174 @@ public Issue GetIssue()
...
+            Context.Subscriptions.Add(subscription);
+            Context.SaveChanges();
+            Repository repository = new Repository();
+            GithubClient.Setup(x => x.Repository.Get(It.IsAny<string>(), It.IsAny<string>())).ReturnsAsync(repository);
+            Issue updateIssue = GetIssue();
+            Octokit.AuthorAssociation author = new Octokit.AuthorAssociation();
+            IssueComment comment = new IssueComment
+            (
+                1,
+                null,
```
Comment (ChadNedzlek) [inline src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/UpdateGithubIssueTest.cs:169]: Add parameter names to the arguments here, I don't know what the 6 nulls are supposed to be.
